MattermostBot/MattermostApiAdapter.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using ApiClient;
+using MattermostApi;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace ApiAdapter
+{
+  /// <summary>
+  /// Класс адаптера для Mattermost. 
+  /// </summary>
+  public class MattermostApiAdapter : IApiClient
+  {
+    private readonly Api api;
+
+    private Action<MessageEventInfo> EventHandler;
+
+    /// <summary>
+    /// Тело запроса добавления эмодзи.
+    /// </summary>
+    private class AddReactionRequestBody
+    {
+      public string user_id;
+      public string post_id;
+      public string emoji_name;
+      public Int64 create_at;
+    }
+
+    /// <summary>
+    /// Сообщение с веб-сокета сервера.
+    /// </summary>
+    private class ServerWebSocketMessage
+    {
+      public string @event;
+      public MessageData data;
+    }
+
+    /// <summary>
+    /// Данные сообщения от веб-сокета.
+    /// </summary>
+    private class MessageData
+    {
+      public string post;
+    }
+
+    /// <summary>
+    /// Класс события о сообщении. 
+    /// </summary>
+    private class PostData
+    {
+      public string id;
+      public string message;
+      public string channel_id;
+      public string user_id;
+      public string root_id;
+    }
+
+    public Message[] GetPinnedMessages(string channelID)
+    {
+      return GetMessagesByRequest(Api.Combine("channels", channelID, "pinned"));
+    }
+
+    public void PostMessage(string channelID, string message, string rootId = null)
+    {
+      Post.Create(api, channelID, message, rootId);
+    }
+
+    public Message[] GetThreadMessages(string postID)
+    {
+      return GetMessagesByRequest(Api.Combine("posts", postID, "thread"));
+    }
+
+    public void PinMessage(string messageID)
+    {
+      var post = Post.GetById(api, messageID).Result;
+      post.Pin(api);
+    }
+
+    public void UnpinMessage(string messageID)
+    {
+      var post = Post.GetById(api, messageID).Result;
+      post.Unpin(api);
+    }
+
+    public void AddReaction(string userId, string messageID, string emodjiName)
+    {
+      api.PostAsync("reactions", null, 
+        new AddReactionRequestBody() { user_id = userId, post_id = messageID, emoji_name = emodjiName, create_at = 0 });
```

Comment:
Можно использовать анонимный тип, т.к. класс `AddReactionRequestBody` больше нигде не используется.

```suggestion
      api.PostAsync("reactions", null, 
        new { user_id = userId, post_id = messageID, emoji_name = emodjiName, create_at = 0 });
```

Reply:
Удалила класс AddReactionRequestBody.